### ISSUE TITLE
Save / Restore not working for voodoo1

### ISSUE THREAD

**Issue by luibgn** [bochs-emu/Bochs#385]
Bochs freezes in the restore process.
Voodoo2 is ok.

**Comment by luibgn** [comment]
I think that the voodoo thread is waiting for the wakeup semaphore but something is wrong in the vertical timer or update timing methods so the thread sleeps forever.
I also think that the semaphores should be created in the init method and not in the thread start method.
And there are probably other small things to fix.

**Comment by Vort** [comment]
> I think that the voodoo thread is waiting for the wakeup semaphore

Main thread waits on this line:
https://github.com/bochs-emu/Bochs/blob/ae7b67242fd896b84e943e0bb25a6ee1afad86a6/bochs/iodev/display/voodoo_func.h#L1166
<details><summary>with such stack</summary>

```
 	bochs.exe!bx_wait_sem(bx_thread_sem_t * thread_sem) Line 101	C++
>	bochs.exe!swapbuffer(_voodoo_state * v, unsigned int data) Line 1172	C++
 	bochs.exe!register_w(unsigned int offset, unsigned int data, bool log) Line 2101	C++
 	bochs.exe!register_w_common(unsigned int offset, unsigned int data) Line 3570	C++
 	bochs.exe!voodoo_w(unsigned int offset, unsigned int data, unsigned int mask) Line 3794	C++
 	bochs.exe!bx_voodoo_1_2_c::mem_write(unsigned __int64 addr, unsigned int len, void * data) Line 871	C++
 	bochs.exe!bx_voodoo_1_2_c::mem_write_handler(unsigned __int64 addr, unsigned int len, void * data, void * param) Line 847	C++
 	bochs.exe!BX_MEM_C::writePhysicalPage(BX_CPU_C * cpu, unsigned __int64 addr, unsigned int len, void * data) Line 76	C++
 	bochs.exe!BX_CPU_C::access_write_physical(unsigned __int64 paddr, unsigned int len, void * data) Line 2571	C++
 	bochs.exe!BX_CPU_C::access_write_linear(unsigned __int64 laddr, unsigned int len, unsigned int curr_pl, unsigned int xlate_rw, unsigned int ac_mask, void * data) Line 478	C++
 	bochs.exe!BX_CPU_C::write_linear_dword(unsigned int s, unsigned __int64 laddr, unsigned int data) Line 102	C++
 	bochs.exe!BX_CPU_C::write_virtual_dword_32(unsigned int s, unsigned int offset, unsigned int data) Line 46	C++
 	bochs.exe!BX_CPU_C::MOV32_EdGdM(bxInstruction_c * i) Line 48	C++
 	bochs.exe!BX_CPU_C::MOV32_GdEdM(bxInstruction_c * i) Line 75	C++
 	bochs.exe!BX_CPU_C::OR_GdEdR(bxInstruction_c * i) Line 180	C++
 	bochs.exe!BX_CPU_C::ADD_GdEdR(bxInstruction_c * i) Line 74	C++
 	bochs.exe!BX_CPU_C::AND_EdIdR(bxInstruction_c * i) Line 268	C++
 	bochs.exe!BX_CPU_C::SETNLE_EbR(bxInstruction_c * i) Line 301	C++
 	bochs.exe!BX_CPU_C::TEST_EdGdR(bxInstruction_c * i) Line 281	C++
 	bochs.exe!BX_CPU_C::cpu_loop() Line 123	C++
 	bochs.exe!bx_begin_simulation(int argc, char * * argv) Line 1051	C++
 	bochs.exe!bx_real_sim_c::begin_simulation(int argc, char * * argv) Line 971	C++
 	bochs.exe!win32_ci_callback(void * userdata, ci_command_t command) Line 778	C++
 	bochs.exe!bx_real_sim_c::configuration_interface(const char * ignore, ci_command_t command) Line 939	C++
 	bochs.exe!bxmain() Line 336	C++
 	bochs.exe!main(int argc, char * * argv) Line 555	C++
```
</details> 

FIFO thread waits on this line:
https://github.com/bochs-emu/Bochs/blob/ae7b67242fd896b84e943e0bb25a6ee1afad86a6/bochs/iodev/display/voodoo.cc#L186

Here is similar problem: https://github.com/bochs-emu/Bochs/issues/146#issuecomment-1931555914.

**Comment by vruppert** [comment]
Please try again the latest Bochs code. The FBI FIFO was missing in save/restore handling.

**Comment by Vort** [comment]
My test shows that problem is fixed with abe3a6f.
Let's see what @luibgn will say.

**Comment by luibgn** [comment]
It's ok. Thanks Ruppert. Good work as usual.
